addons/loyalty/models/product_pricelist.py

```diff
@@ -0,0 +1,20 @@
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class ProductPricelist(models.Model):
+    _inherit = 'product.pricelist'
+
+    def action_archive(self):
+        loyalty_programs = self.env['loyalty.program'].search([
+            ('active', '=', True),
+            ('pricelist_ids', 'in', self.ids)
+        ])
+        program_names = [l.name for l in loyalty_programs]
+        if program_names:
```

Comment:
```suggestion
        if loyalty_programs:
```
I don't think you can create programs without names, and even if you could, you should still see the warning.